File: android/app/src/main/java/com/browserselector/android/model/UrlRule.kt
package com.browserselector.android.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

/**
 * Represents a URL pattern rule that maps URLs to specific browsers.
 */
@Entity(
    tableName = "url_rules",
    foreignKeys = [
        ForeignKey(
            entity = Browser::class,
            parentColumns = ["packageName"],
            childColumns = ["browserPackage"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index("browserPackage")]
)
data class UrlRule(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,

    val pattern: String,

    val browserPackage: String,

    val priority: Int = 0,

    val createdAt: Long = System.currentTimeMillis()
) {
    /**
     * Creates a copy with updated priority.
     */
    fun withPriority(priority: Int): UrlRule = copy(priority = priority)
}

File: ios/BrowserSelector/Models/UrlRule.swift
import Foundation
import SwiftData

/// Represents a URL pattern rule that maps URLs to specific browsers.
@Model
class UrlRule {
    var pattern: String
    var browserScheme: String
    var priority: Int
    var createdAt: Date

    init(pattern: String, browserScheme: String, priority: Int = 0) {
        self.pattern = pattern
        self.browserScheme = browserScheme
        self.priority = priority
        self.createdAt = Date()
    }

    /// Creates a copy with updated priority.
    func withPriority(_ priority: Int) -> UrlRule {
        UrlRule(pattern: pattern, browserScheme: browserScheme, priority: priority)
    }
}
